### PR TITLE
perf: improve pc/ic mapping stuff

### PR DESCRIPTION
**Pull request by onbjerg**
## Motivation

After enabling fuzz tests in coverage (no data is used yet) coverage went from taking about 2 seconds for solmate to 27 seconds. The hottest piece of code seems to be the coverage collector itself that does some very expensive computations (translating program counters to instruction counters and then recording them) on every call.

## Solution

Instead of doing the PC -> IC translation in the inspector, we can do it once before or after we collect hit data. The hit data is then just a mapping of PC -> hits. This took the time back down to 2 seconds.

For coverage I opted to just move from using instruction counters in the anchors to program counters, so we use an IC -> PC map (since source maps are based on instruction counters, we need to translate from instruction counters to program counters in some cases).

I'll look into doing the same for the debugger, because it uses the same translation mechanism

### CODE REVIEW

**Comment by gakonst** [comment]
is there something blocking us here?

**Comment by onbjerg** [comment]
no, will get it over the line today, just want to do the same for debugging and have been focusing on the solc AST PR

File: ui/src/lib.rs [hunk 1]
```diff
@@ -75,6 +77,30 @@ impl Tui {
         let backend = CrosstermBackend::new(stdout);
         let mut terminal = Terminal::new(backend)?;
         terminal.hide_cursor();
+        let pc_ic_maps = known_contracts
```
Comment (mattsse) [inline ui/src/lib.rs:80]: we really need to turn this know_contracts Btreemap thing into a new type -.-

**Review by mattsse — APPROVED** [review]
lgtm

File: ui/src/lib.rs (hunk 1 above)
Comment (onbjerg) [reply ui/src/lib.rs:80]: yes, or find another way to get rid of it :smile: will be easier when things like #2692 is done so it's easier to reason about, i don't remember other places this is used

**Pull request merged**